packages/server/lib/util/validation.js

```diff
@@ -104,18 +104,19 @@ const isValidBrowserList = (key, browsers) => {
 }
 
 const isValidRetriesConfig = (key, value) => {
-  const isNullOrNumber = isOneOf([_.isNumber, _.isNull])
-
-  if (
-    isNullOrNumber(value)
-    || (_.isEqual(_.keys(value), ['runMode', 'openMode']))
-      && isNullOrNumber(value.runMode)
-      && isNullOrNumber(value.openMode)
-  ) {
+  const optionalKeys = ['runMode', 'openMode']
+  const isValidRetryValue = (val) => _.isNull(val) || (Number.isInteger(val) && val >= 0)
+  const optionalKeysAreValid = (val, k) => optionalKeys.includes(k) && isValidRetryValue(val)
+
+  if (isValidRetryValue(value)) {
+    return true
+  }
+
+  if (_.isObject(value) && _.every(value, optionalKeysAreValid)) {
     return true
   }
 
-  return errMsg(key, value, 'a number or null or an object with keys "openMode" and "runMode" with values of numbers or nulls')
+  return errMsg(key, value, 'a positive number or null or an object with keys "openMode" and "runMode" with values of numbers or nulls')
```

Comment:
Think this would read better, with commas instead of "this or this or this"

```suggestion
  return errMsg(key, value, 'a positive number, null, or an object with keys "openMode" and "runMode" with a positive number or null values')
```

Reply:
I agree but that’s not our style in other error messages.

Reply:
Added an internal story TR-239